t/mojo/delay.t

```diff
@@ -6,197 +6,211 @@ use Test::More;
+      Mojo::IOLoop->next_tick(sub { $end->(1, 2, 3)->pass(5) });
+    },
+    sub {
+      my ($delay, @numbers) = @_;
+      my $end = $delay->begin;
+      Mojo::IOLoop->next_tick(sub { $end->(undef, @numbers, 4) });
+    },
+    sub {
+      my ($delay, @numbers) = @_;
+      $result = \@numbers;
+    }
+  )->wait;
+  is_deeply $result, [2, 3, 2, 1, 4, 5], 'right results';
+};
+
+subtest 'One step' => sub {
+  my $result;
+  my $delay = Mojo::IOLoop::Delay->new;
+  $delay->steps(sub { ++$result });
+  $delay->begin->();
+  is $result, undef, 'no result';
+  Mojo::IOLoop->next_tick($delay->begin);
+  is $result, undef, 'no result';
+  my $end = $delay->begin;
+  Mojo::IOLoop->next_tick(sub { $end->() });
+  is $result, undef, 'no result';
+  $delay->wait;
+  is $result, 1, 'right result';
+};
+
+subtest 'One step (reverse)' => sub {
+  my $result;
+  my $delay = Mojo::IOLoop::Delay->new;
+  my $end   = $delay->begin(0);
+  Mojo::IOLoop->next_tick(sub { $end->(23) });
+  $delay->steps(sub { $result = pop });
+  is $result, undef, 'no result';
+  $delay->wait;
+  is $result, 23, 'right result';
+};
+
+subtest 'End chain after first step' => sub {
+  my $result;
+  my $delay = Mojo::IOLoop::Delay->new;
+  $delay->steps(sub { $result = 'success' }, sub { $result = 'fail' });
+  $delay->wait;
+  is $result, 'success', 'right result';
+};
+
+subtest 'End chain after third step' => sub {
+  my $result;
+  my $delay = Mojo::IOLoop::Delay->new;
+  $delay->steps(
+    sub { Mojo::IOLoop->next_tick(shift->begin) },
+    sub {
+      $result = 'fail';
+      shift->pass;
+    },
+    sub { $result = 'success' },
+    sub { $result = 'fail' }
+  )->wait;
+  is $result, 'success', 'right result';
+};
+
+subtest 'End chain after second step' => sub {
+  my @results;
+  my $delay = Mojo::IOLoop::Delay->new;
+  $delay->then(sub { push @results, [@_] });
+  $delay->steps(sub { shift->pass(23) }, sub { shift; push @results, [@_] }, sub { push @results, 'fail' })->wait;
+  is_deeply \@results, [[23], [23]], 'right results';
+};
+
+subtest 'Nested delays' => sub {
+  my $result;
+  my $delay = Mojo::IOLoop->delay(
+    sub {
+      my $first  = shift;
+      my $second = Mojo::IOLoop->delay($first->begin);
+      Mojo::IOLoop->next_tick($second->begin);
+      Mojo::IOLoop->next_tick($first->begin);
+      my $end = $second->begin(0);
+      Mojo::IOLoop->next_tick(sub { $end->(1, 2, 3) });
+    },
+    sub {
+      my ($first, @numbers) = @_;
+      $result = \@numbers;
+      my $end = $first->begin;
+      $first->begin->(3, 2, 1);
+      my $end2 = $first->begin(0);
+      my $end3 = $first->begin(0);
+      $end2->(4);
+      $end3->(5, 6);
+      $first->pass(23)->pass(24);
+      $end->(1, 2, 3);
+    },
+    sub {
+      my ($first, @numbers) = @_;
+      push @$result, @numbers;
+    }
+  )->wait;
+  is_deeply $result, [1, 2, 3, 2, 3, 2, 1, 4, 5, 6, 23, 24], 'right results';
+};
+
+subtest 'Exception in first step' => sub {
+  my $failed;
+  my $result;
```

Comment:
On separate lines is inconsistent.

Reply:
Very good, thanks for clarifying that the parentheses are preferred!

Reply:
The main preference is consistency. If something is unclear just look at how other files are formatted.